native/common/jp_proxy.cpp

```diff
@@ -93,7 +101,7 @@ extern "C" JNIEXPORT jobject JNICALL Java_org_jpype_proxy_JPypeProxy_hostInvoke(
 
 			// convert the arguments into a python list
 			JP_TRACE("Convert arguments");
-			JPPyObject pyargs = getArgs(context, parameterTypePtrs, args);
+			JPPyObject pyargs = getArgs(context, parameterTypePtrs, args, proxy->m_Instance->m_Target, addSelf);
```

Comment:
Is m_Instance always valid?

Reply:
It is being initialized in the ctor, but never checked. This could cause a crash.

Reply:
If m_Instance was null we would have big problems because that was part of the object that invoked it.  m_Target also can't be null or we would have already crashed Python on the clear method.

Locations it is called from  pyjp_proxy.cpp
```
     self->m_Proxy = new JPProxyDirect(context, self, interfaces);
```
jp_functional.cpp
```
        self->m_Proxy = new JPProxyFunctional(context, self, cl);
```

I could add a check, but I would never be able to exercise such a check and we would have crashed by construction given the value passing in is the reference to the Python object which is constructing it.